### PR TITLE
Fix failing pre-commits after making 3.7 default

### PR DESCRIPTION
**Pull request by potiuk**
In order to keep consistency, both mypy and flake were always
using Python 3.6 - simply to make sure that the same errors
are produced for everyone. When default python version was
changed in #19317 the regular PR static checks started to fail
because Python 3.6 image is no longer built for "regular" PRs
(it's only built when "full-tests-needed" is set or when
the build runs in "main" as result of push or schedule).

Unfortunately those "defaults" could not be read from
"default python version" easily, because they are also
executed in pre-commits, which do not have the same
initialization as `breeze` or other CI scripts so there
are literally 3 more places where Python 3.6 should be
changed to 3.7.

This PR fixes it.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
This should fix the 2nd (and last widespread) failure in regular PRs after chaning 3.7 to be default :).

**Comment by potiuk** [comment]
Hmm. Seems like running current mypy with 3.7 ACTUALLY causes problems:

```
/usr/local/lib/python3.7/site-packages/git/cmd.py:71: error: INTERNAL ERROR -- Please try using mypy master on Github:
```

**Comment by potiuk** [comment]
All right - I feel like good opportunity to migrate to latest mypy 😱 . 

We had it fixed to 0.770 which was released 11th May 2020. Sounds like we are **little** behind.

**Comment by potiuk** [comment]
CC: @dstandish -> I am not sure how Many problems we will have now with moving mypy from 0.770 to 0.910, but imagine if every developer had their own version of mypy installation with their own versions of  other dependencies in their own version of Python. 

This is is why running mypy in the same docker with the same python + the same dependencies is a very good option (even if it is slower on MacOS) :(.

In this case just changing Image from 3.6 to 3.7 to run mypy caused:

```
/usr/local/lib/python3.7/site-packages/git/cmd.py:71: error: INTERNAL ERROR -- Please try using mypy master on Github:
```

But I expect quite a number of new errors reported with newer version of Mypy.

**Comment by potiuk** [comment]
BTW. I am waiting (two weeks) for my new McBook Pro (Late 2021 - with `notch` y'know ) - so I hope I will be able to take a closer look and wit this one `stone` (it's much heavier and bulkier than previous ones) kill several birds which were kinda` flying loose:

* Docker Image for ARM (BTW M1 could be another reason for slow-downs you saw - so far our images are x86 only and on M1 they use emulation - that's why they can be WAY slower)
* Speedup some docker operations on MacOS - mypy for one
* Finally try some of the new (ish) features in docker stack (mostly BuildKit - which I think will be much easier when it comes to making  multi-platform images)

**Comment by dstandish** [comment]
> BTW. I am waiting (two weeks) for my new McBook Pro (Late 2021 - with `notch` y'know ) 

very exciting.  and good luck to you 😅

**Review by dstandish — APPROVED** [review]

**Comment by uranusjr** [comment]
Merging into #19333

**Comment by potiuk** [comment]
We still need to fix some resulting mypy errors @uranusjr  I am afraid  :)

**Comment by potiuk** [comment]
Als this one is really separate from fixing the warnings (I will try to fix the mypy errors today unless someone bits me to it).

**Comment by potiuk** [comment]
I will not be near my PC whole day today - but maybe someone could simply comment out the failing mypy precommits temporarily until we fix it ?

**Comment by potiuk** [comment]
Added PR to remove mypy checks temporarily https://github.com/apache/airflow/pull/19345

**Review by kaxil — APPROVED** [review]

File: airflow/api_connexion/endpoints/dag_endpoint.py [hunk 1]
```diff
@@ -111,15 +110,15 @@ def patch_dag(session, dag_id, update_mask=None):
 
 @security.requires_access([(permissions.ACTION_CAN_DELETE, permissions.RESOURCE_DAG)])
 @provide_session
-def delete_dag(dag_id: str, session: Session):
+def delete_dag(dag_id: str, session):
```
Comment (uranusjr) [inline airflow/api_connexion/endpoints/dag_endpoint.py:113]: I don’t think this is supposed to fail?

File: airflow/cli/simple_table.py [hunk 2]
```diff
@@ -70,13 +70,13 @@ def print_as_plain_table(self, data: List[Dict]):
             self.print("No data found")
             return
         rows = [d.values() for d in data]
-        output = tabulate(rows, tablefmt="plain", headers=data[0].keys())
+        output = tabulate(rows, tablefmt="plain", headers=list(data[0].keys()))
```
Comment (uranusjr) [inline airflow/cli/simple_table.py:73]:
```suggestion
        output = tabulate(rows, tablefmt="plain", headers=list(data[0]))
```

File: airflow/config_templates/default_webserver_config.py [hunk 3]
```diff
@@ -18,12 +18,12 @@
...
+from airflow.www.fab_security.manager import AUTH_DB  # type: ignore
 
-# from airflow.www.fab_security.manager import AUTH_LDAP
-# from airflow.www.fab_security.manager import AUTH_OAUTH
-# from airflow.www.fab_security.manager import AUTH_OID
-# from airflow.www.fab_security.manager import AUTH_REMOTE_USER
+# from airflow.www.fab_security.manager import AUTH_LDAP  # type: ignore
+# from airflow.www.fab_security.manager import AUTH_OAUTH  # type: ignore
+# from airflow.www.fab_security.manager import AUTH_OID  # type: ignore
+# from airflow.www.fab_security.manager import AUTH_REMOTE_USER  # type: ignore
```
Comment (uranusjr) [inline airflow/config_templates/default_webserver_config.py:26]: Why?

File: airflow/kubernetes/kubernetes_helper_functions.py [hunk 4]
```diff
@@ -84,4 +84,8 @@ def annotations_to_key(annotations: Dict[str, str]) -> Optional[TaskInstanceKey]
             .scalar()
         )
 
+    if not run_id:
+        # TODO: Is this the right way of handling it?
+        raise ValueError("Run Id has to be defined either by `run_id` annotation "
+                         "or derived from `execution_Date")
```
Comment (uranusjr) [inline airflow/kubernetes/kubernetes_helper_functions.py:90]: We can also just do

```python
(run_id,) = (
    session.query(TaskInstance.run_id)
    ...
    .one()
)
```

and let SQLAlchemy raise when there’s not a matching task instance found (it shouldn’t happen).

**Review by uranusjr — COMMENTED** [review]
Didn’t read everything but there are some things we can do a bit better

**Comment by potiuk** [comment]
> Didn’t read everything but there are some things we can do a bit better

Yeah. It's a draft - I want to make all the changes first and split it into a series of "targetted" PRs (otherwise it will be impossible to review) and re-enable mypy after I merge them all.

**Comment by potiuk** [comment]
MyPy effort is now spread among MANY people :)

**Pull request closed without merging**